privacyidea/lib/policy.py

```diff
@@ -618,6 +619,31 @@ def list_policies(self, name=None, scope=None, realm=None, active=None,
                 log.debug("Policies after matching {1!s}={2!s}: {0!s}".format(
                     reduced_policies, searchkey, searchvalue))
 
+        for searchkey, searchvalue in q:
+            if searchvalue is not None:
+                new_policies = []
+                # first we find policies, that really match!
+                # Either with the real value or with a "*"
+                # values can be excluded by a leading "!" or "-"
+                for policy in reduced_policies:
+                    if not policy.get(searchkey):
+                        # We also find the policies with no distinct information
+                        # about the request value
+                        new_policies.append(policy)
+                    elif policy.get("user_case_insensitive"):
+                        value_found, value_excluded = self._search_value(
+                            [x.lower() for x in policy.get(searchkey)], searchvalue.lower())
+                        if value_found and not value_excluded:
+                            new_policies.append(policy)
+                    else:
+                        value_found, value_excluded = self._search_value(
+                            policy.get(searchkey), searchvalue)
+                        if value_found and not value_excluded:
+                            new_policies.append(policy)
```

Comment:
Having an *elif* and an *else* branch is somehow confusing, because in both branches the same thing happens.

I would recommend the following:

~~~~Python
else:
    searchkeys = policy.get(searchkey)
    if policy.get("user_case_insensitive"):
        searchvalue = searchvalue.lower
        searchkeys =  [x.lower() for x in searchkeys]
    value_found, value_excluded = self._search_value(
                            searchkeys, searchvalue)
    if value_found and not value_excluded:
        new_policies.append(policy)
~~~~

Reply:
This can cause an issue if the search value is not already lowercase